sdk/vppagent/flush.go

```diff
@@ -0,0 +1,134 @@
+
+	if f.GetNext() == nil {
+		logrus.Fatal("The VPP Agent Flush composite requires that there is Next set")
+	}
+
+	if ctx == nil {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(context.Background(), 120*time.Second)
+		defer cancel()
+	}
+	if err := tools.WaitForPortAvailable(ctx, "tcp", f.Endpoint, 100*time.Millisecond); err != nil {
+		return nil, err
+	}
+
+	tracer := opentracing.GlobalTracer()
+	conn, err := grpc.Dial(f.Endpoint, grpc.WithInsecure(),
+		grpc.WithUnaryInterceptor(
+			otgrpc.OpenTracingClientInterceptor(tracer, otgrpc.LogPayloads())),
+		grpc.WithStreamInterceptor(
+			otgrpc.OpenTracingStreamClientInterceptor(tracer)))
+
+	if err != nil {
+		logrus.Errorf("Can't dial grpc server: %v", err)
+		return nil, err
+	}
+	defer func() { _ = conn.Close() }()
+	client := configurator.NewConfiguratorClient(conn)
+
+	incomingConnection, err := f.GetNext().Request(ctx, request)
+	if err != nil {
+		logrus.Errorf("Next request failed: %v", err)
+		return nil, err
+	}
+
+	opaque := f.GetNext().GetOpaque(incomingConnection)
+	if opaque == nil {
+		err := fmt.Errorf("received empty data from Next")
+		logrus.Errorf("Unable to find the DataChange: %v", err)
+		return nil, err
+	}
+	dataChange := opaque.(*ConnectionData).DataChange
+
+	logrus.Infof("Sending DataChange to VPP Agent: %v", dataChange)
+	if _, err := client.Update(ctx, &configurator.UpdateRequest{Update: dataChange}); err != nil {
+		logrus.Error(err)
+		_, _ = client.Delete(ctx, &configurator.DeleteRequest{Delete: dataChange})
+		return nil, err
+	}
+
+	return incomingConnection, nil
+}
+
+func (f *Flush) reset() error {
+	ctx, cancel := context.WithTimeout(context.Background(), 120*time.Second)
+	defer cancel()
+	if err := tools.WaitForPortAvailable(ctx, "tcp", f.Endpoint, 100*time.Millisecond); err != nil {
+		return err
+	}
+	tracer := opentracing.GlobalTracer()
+	conn, err := grpc.Dial(f.Endpoint, grpc.WithInsecure(),
+		grpc.WithUnaryInterceptor(
+			otgrpc.OpenTracingClientInterceptor(tracer, otgrpc.LogPayloads())),
+		grpc.WithStreamInterceptor(
+			otgrpc.OpenTracingStreamClientInterceptor(tracer)))
+
+	if err != nil {
+		logrus.Errorf("Can't dial grpc server: %v", err)
+		return err
+	}
+	defer func() { _ = conn.Close() }()
+	client := configurator.NewConfiguratorClient(conn)
+	logrus.Infof("Resetting VPP Agent...")
+	_, err = client.Update(context.Background(), &configurator.UpdateRequest{
+		Update:     &configurator.Config{},
+		FullResync: true,
+	})
+	if err != nil {
+		logrus.Errorf("Failed to reset VPP Agent: %s", err)
+	}
+	logrus.Infof("Finished resetting VPP Agent")
+	return nil
+}
+
+// Close implements the close handler
+func (f *Flush) Close(ctx context.Context, connection *connection.Connection) (*empty.Empty, error) {
+	if f.GetNext() != nil {
+		return f.GetNext().Close(ctx, connection)
+	}
+	return &empty.Empty{}, nil
+}
+
+// NewFlush creates a Flush
+func NewFlush(configuration *common.NSConfiguration, endpoint string) *Flush {
```

Comment:
Naming is hard, can we have a better name here? Maybe Update?

Reply:
For this func only? Not for the type, right? I thought there is some kind of name convention for these functions. Should I rename other New* functions in vppagent package too?

Reply:
Oh my fault, should have been more clear. `New*` is the standart, that is OK. WHat I meant was `flush` :) So like `NewUpdate` although that sounds weird too. But in any case we probably can figure out a better name than `flush`. Unless of course that is a vppagent terminology?

Reply:
Not sure it is VPP Agent terminology, but seems `flush` is usually used to write some buffered data, it is exactly what our composite does.

@edwarnicke maybe you can explain more on this name choice?

Reply:
Yeah, name is not good. What about NewVppUpdateRequest?

Reply:
I'm fine with a better name than 'Flush'... I've been thinking a lot lately about storage, so "flush to disk" is bouncing around in my head.  NewVppUpdate or NewVPPUpdateRequest or NewSendVppUpdate or NewProgramVPP etc would all likely be better than Flush :)
@nickolaev @haiodo @fkautz thoughts on naming?

I do have a question though... would it be OK to fix the naming in a follow up PR?  I'm liking what I'm seeing so far in the PR, and so would love to (if the rest looks good) merge it... :)